sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -226,9 +224,7 @@ protected void initialize(String path, List<String> columns) throws IOException
     this.reader = new ParquetFileReader(
         config, footer.getFileMetaData(), file, blocks, requestedSchema.getColumns());
     // use the blocks from the reader in case some do not match filters and will not be read
-    for (BlockMetaData block : reader.getRowGroups()) {
-      this.totalRowCount += block.getRowCount();
-    }
+    this.totalRowCount += reader.getFilteredRecordCount();
```

Comment:
The latest benchmark result:


Case | Parquet 1.11 Vectorized(ms) | Parquet 1.11 Vectorized(Pushdown)(ms) | Parquet 1.10 Vectorized(ms) | Parquet 1.10   Vectorized(Pushdown)(ms)
--- | --- | --- | --- | ---
Select 0 string row (value IS NULL) | 7001 | 631 | 8459 | 569
Select 0 string row ('7864320' < value   < '7864320') | 8801 | 744 | 9596 | 470
Select 1 string row (value = '7864320') | 6973 | 578 | 8415 | 456
Select 1 string row (value <=>   '7864320') | 7090 | 867 | 9681 | 663
Select 1 string row ('7864320' <=   value <= '7864320') | 7637 | 639 | 8257 | 442
Select all string rows (value IS NOT   NULL) | 14638 | 14926 | 15058 | 17091
Select 0 int row (value IS NULL) | 7233 | 532 | 8373 | 460
Select 0 int row (7864320 < value <   7864320) | 6474 | 558 | 8176 | 620
Select 1 int row (value = 7864320) | 7284 | 554 | 7545 | 435
Select 1 int row (value <=>   7864320) | 7109 | 724 | 8550 | 484
Select 1 int row (7864320 <= value   <= 7864320) | 6340 | 563 | 7648 | 440
Select 1 int row (7864319 < value <   7864321) | 7134 | 620 | 7521 | 435
Select 10% int rows (value < 1572864) | 7561 | 1986 | 8790 | 1988
Select 50% int rows (value < 7864320) | 10425 | 7434 | 10445 | 7133
Select 90% int rows (value < 14155776) | 12130 | 11745 | 12959 | 12574
Select all int rows (value IS NOT NULL) | 12662 | 12961 | 13640 | 13794
Select all int rows (value > -1) | 12568 | 12864 | 13547 | 13691
Select all int rows (value != -1) | 12574 | 12874 | 14617 | 14533
Select 0 distinct string row (value IS   NULL) | 5925 | 455 | 7013 | 371
Select 0 distinct string row ('100' <   value < '100') | 6037 | 445 | 7087 | 391
Select 1 distinct string row (value =   '100') | 6107 | 603 | 7169 | 524
Select 1 distinct string row (value   <=> '100') | 6309 | 1418 | 7113 | 528
Select 1 distinct string row ('100' <=   value <= '100') | 6224 | 620 | 7222 | 549
Select all distinct string rows (value IS   NOT NULL) | 14198 | 14293 | 15175 | 16194
StringStartsWith filter: (value like   '10%') | 8399 | 3572 | 10298 | 2642
StringStartsWith filter: (value like   '1000%') | 7424 | 559 | 7998 | 441
StringStartsWith filter: (value like   '786432%') | 7554 | 542 | 7920 | 428
Select 1 decimal(9, 2) row (value =   7864320) | 2684 | 131 | 3834 | 115
Select 10% decimal(9, 2) rows (value <   1572864) | 4201 | 2280 | 5139 | 2170
Select 50% decimal(9, 2) rows (value <   7864320) | 8661 | 8325 | 9593 | 10449
Select 90% decimal(9, 2) rows (value <   14155776) | 10213 | 9833 | 11647 | 11828
Select 1 decimal(18, 2) row (value =   7864320) | 3259 | 150 | 4631 | 133
Select 10% decimal(18, 2) rows (value   < 1572864) | 4072 | 1284 | 5285 | 1260
Select 50% decimal(18, 2) rows (value   < 7864320) | 7010 | 5495 | 7959 | 5898
Select 90% decimal(18, 2) rows (value   < 14155776) | 10037 | 9957 | 10845 | 10535
Select 1 decimal(38, 2) row (value =   7864320) | 4970 | 151 | 5943 | 131
Select 10% decimal(38, 2) rows (value   < 1572864) | 5912 | 1605 | 7079 | 1827
Select 50% decimal(38, 2) rows (value   < 7864320) | 9784 | 7573 | 11497 | 7991
Select 90% decimal(38, 2) rows (value   < 14155776) | 13935 | 13341 | 14702 | 14183
InSet -> InFilters (values count: 5,   distribution: 10) | 7193 | 600 | 8001 | 495
InSet -> InFilters (values count: 5,   distribution: 50) | 7002 | 577 | 8042 | 480
InSet -> InFilters (values count: 5,   distribution: 90) | 7003 | 587 | 8526 | 484
InSet -> InFilters (values count: 10,   distribution: 10) | 6984 | 625 | 8279 | 519
InSet -> InFilters (values count: 10,   distribution: 50) | 6949 | 706 | 8097 | 505
InSet -> InFilters (values count: 10,   distribution: 90) | 7336 | 613 | 7961 | 507
InSet -> InFilters (values count: 50,   distribution: 10) | 7369 | 7475 | 8052 | 8244
InSet -> InFilters (values count: 50,   distribution: 50) | 7295 | 7619 | 8202 | 8311
InSet -> InFilters (values count: 50,   distribution: 90) | 7584 | 7610 | 8405 | 8326
InSet -> InFilters (values count: 100,   distribution: 10) | 7264 | 7358 | 8041 | 8200
InSet -> InFilters (values count: 100,   distribution: 50) | 7192 | 7277 | 8019 | 8437
InSet -> InFilters (values count: 100,   distribution: 90) | 7040 | 7236 | 10567 | 10681
Select 1 tinyint row (value = CAST(63 AS   tinyint)) | 3185 | 247 | 4855 | 235
Select 10% tinyint rows (value <   CAST(12 AS tinyint)) | 3823 | 1120 | 5091 | 1209
Select 50% tinyint rows (value <   CAST(63 AS tinyint)) | 6570 | 5117 | 9265 | 6076
Select 90% tinyint rows (value <   CAST(114 AS tinyint)) | 9291 | 9229 | 10508 | 10152
Select 1 timestamp stored as INT96 row   (value = CAST(7864320 AS timestamp)) | 4054 | 4757 | 6253 | 4774
Select 10% timestamp stored as INT96 rows   (value < CAST(1572864 AS timestamp)) | 6190 | 4512 | 5295 | 13923
Select 50% timestamp stored as INT96 rows   (value < CAST(7864320 AS timestamp)) | 8681 | 7207 | 9758 | 8633
Select 90% timestamp stored as INT96 rows   (value < CAST(14155776 AS timestamp)) | 13536 | 9738 | 11053 | 11642
Select 1 timestamp stored as   TIMESTAMP_MICROS row (value = CAST(7864320 AS timestamp)) | 2904 | 136 | 4279 | 120
Select 10% timestamp stored as   TIMESTAMP_MICROS rows (value < CAST(1572864 AS timestamp)) | 3983 | 1209 | 4872 | 1251
Select 50% timestamp stored as   TIMESTAMP_MICROS rows (value < CAST(7864320 AS timestamp)) | 6886 | 6052 | 7441 | 5546
Select 90% timestamp stored as   TIMESTAMP_MICROS rows (value < CAST(14155776 AS timestamp)) | 9097 | 10715 | 10205 | 10035
Select 1 timestamp stored as   TIMESTAMP_MILLIS row (value = CAST(7864320 AS timestamp)) | 3278 | 218 | 4349 | 119
Select 10% timestamp stored as   TIMESTAMP_MILLIS rows (value < CAST(1572864 AS timestamp)) | 3955 | 1176 | 6646 | 1276
Select 50% timestamp stored as   TIMESTAMP_MILLIS rows (value < CAST(7864320 AS timestamp)) | 6684 | 5161 | 7766 | 5635
Select 90% timestamp stored as   TIMESTAMP_MILLIS rows (value < CAST(14155776 AS timestamp)) | 10910 | 9297 | 10735 | 10314
Select 1 row with 1 filters | 340 | 351 | 413 | 3317
Select 1 row with 250 filters | 1033 | 1132 | 1256 | 1078
Select 1 row with 500 filters | 3022 | 3353 | 2868 | 3197
Sum | 508182 | 293442 | 579526 | 328587